src/Compiler/Expression.php

```diff
@@ -491,46 +491,40 @@ protected function passSymbol(Node\Expr\Assign $expr)
         $compiledExpression = $this->compile($expr->expr);
 
         if ($expr->var instanceof Node\Expr\List_) {
-            $isCorrectType = false;
+            $isCorrectType = $compiledExpression->isArray();
 
-            switch ($compiledExpression->getType()) {
-                case CompiledExpression::ARR:
-                    $isCorrectType = true;
-                    break;
-            }
+            foreach ($expr->var->vars as $key => $var) {
+                if (!$var instanceof Node\Expr\Variable) {
+                    continue;
+                }
 
-            if ($expr->var->vars) {
-                foreach ($expr->var->vars as $key => $var) {
-                    if ($var instanceof Node\Expr\Variable) {
-                        if (!isset($expr->var->name)) {
-                            var_dump($var);
-                        }
-                        $name = $expr->var->name;
-
-                        $symbol = $this->context->getSymbol($name);
-                        if (!$symbol) {
-                            $symbol = new Variable(
-                                $name,
-                                null,
-                                CompiledExpression::UNKNOWN,
-                                $this->context->getCurrentBranch()
-                            );
-                            $this->context->addVariable($symbol);
-                        }
+                if ($var->name instanceof Node\Expr\Variable) {
+                    // TODO what should we do here?
```

Comment:
Just create the variable. It is the job of the analyzer to notice the usage of variable variables but the compiler should work with it.
The whole passSymbol method has to be refactored later into compiler and analyzer part (and the other pass methods in this class too)